### PR TITLE
V1.4 update UGDG

### PR DESCRIPTION
**Pull request by nbvannhi**
DG implementation details updates
- addalloc
- addscore
- show

Note: I made some cosmetic changes hence it doesn't look standardised currently.
Will update others' part accordingly once everyone is done with their update.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -231,32 +264,67 @@ The following sequence diagram summarizes what happens when the user inputs an a
...
 
-### Add Allocation feature
+### Add Allocation : `addalloc`
+
+The `addalloc` feature allows users to allocate a student into a group.
+
+#### How the `AddAllocCommand` works
 
-The `addalloc` feature allocates an existing student into a group.
+Step 1. The user specifies the group name, the name or ID of the student to be allocated into the group.
```
Comment (jonas-chow) [inline docs/DeveloperGuide.md]: Don't use "Step 1.".
Instead just use the standard numbered list (i.e. 1.)

"Step 1" implies that it's a step that we actively do. In actual fact, it's just the computer doing all the work.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -300,6 +367,36 @@ if command is valid. Command is invalid if user input is empty, or if user enter
     * Pros: More straightforward and convenient for users.
     * Cons: We need to identify the type of input given.
 
+### Show : `show`
+
+The `show` feature allows users to show the performance analysis of a student, a group or the cohort in an assessment.
+
+#### How the `ShowCommand` works
```
Comment (jonas-chow) [inline docs/DeveloperGuide.md]: How the `show` command works

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -231,32 +264,67 @@ The following sequence diagram summarizes what happens when the user inputs an a
...
+![AddGroupSequenceDiagram](images/AddAllocSequenceDiagram.png)
 
-Step 4. `AddAllocCommand#execute()` calls `AddAllocCommand#createEditedStudent()` to create an instance of student `Alex Yeoh` allocated to group `T02`, and modifies the current unallocated student `Alex Yeoh` in the database with the newly allocated instance through `ReadOnlyAddressBook#setStudent`.
+### Add Score : `addscore`
 
-Step 5. `AddAllocCommand#execute()` calls `Group#addStudent()` to add student `Alex Yeoh` into the group.
+The `addscore` feature allows users to add score for an assessment of a student.
 
-Use case ends.
+#### How the `AddScoreCommand` works
```
Comment (jonas-chow) [inline docs/DeveloperGuide.md]: How the `addscore` command works

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -231,32 +264,67 @@ The following sequence diagram summarizes what happens when the user inputs an a
 
 ![AddGroupSequenceDiagram](images/AddGroupSequenceDiagram.png)
 
-### Add Allocation feature
+### Add Allocation : `addalloc`
+
+The `addalloc` feature allows users to allocate a student into a group.
+
+#### How the `AddAllocCommand` works
```
Comment (jonas-chow) [inline docs/DeveloperGuide.md]: How the `addalloc` command works

To be consistent

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -2,21 +2,57 @@
 layout: page
 title: Developer Guide
 ---
-* Table of Contents
-{:toc}
+
+## Table of Contents
```
Comment (jonas-chow) [inline docs/DeveloperGuide.md]: I'm not a fan of changing the table of contents to be not automatically-generated, especially in a long document like this one.

**Review by jonas-chow — CHANGES_REQUESTED** [review]
Looks fine in general, just some minor tweaks should be done in order to stay consistent with the other sections.

I'm generally of the opinion that the sequence diagrams add absolutely no value (especially since they're tiny). Though you can leave those on for now, we can discuss this along with the table of contents later in the meeting

**Comment by nbvannhi** [comment]
> 
> 
> Looks fine in general, just some minor tweaks should be done in order to stay consistent with the other sections.
> 
> I'm generally of the opinion that the sequence diagrams add absolutely no value (especially since they're tiny). Though you can leave those on for now, we can discuss this along with the table of contents later in the meeting

Hm I changed the table of contents to be auto-generated alr (since I also find it inconvenient to have to update it every time something change), but somehow the deployed website doesn't show up as expected

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -231,32 +236,59 @@ The following sequence diagram summarizes what happens when the user inputs an a
...
 ![AddGroupSequenceDiagram](images/AddGroupSequenceDiagram.png)
 
-### Add Allocation feature
+### Add Allocation : `addalloc`
+
+The `addalloc` feature allows users to allocate a student into a group.
+
+#### How the `addalloc` command works
+
+1. The user specifies the group name, the name or ID of the student to be allocated into the group.
```
Comment (jonas-chow) [inline docs/DeveloperGuide.md]: The user specifies the group name, and the name or ID of the student to be allocated into the group.

**Review by jonas-chow — APPROVED** [review]
One small english change I'll fix. Otherwise LGTM.

**Comment by zhing22** [comment]
> > Looks fine in general, just some minor tweaks should be done in order to stay consistent with the other sections.
> > I'm generally of the opinion that the sequence diagrams add absolutely no value (especially since they're tiny). Though you can leave those on for now, we can discuss this along with the table of contents later in the meeting
> 
> Hm I changed the table of contents to be auto-generated alr (since I also find it inconvenient to have to update it every time something change), but somehow the deployed website doesn't show up as expected

i prefer it to be manually written though, the auto generated one looks a bit messy. we just have to make one adjustment at the end after everything else is completed.

**Pull request merged**